File: SOLID/SRP/Violation/bankapp.py
class BankGeneralist:

    def __init__(self):
        self.ledger = []

    # Task 1: Cashier Duty
    def deposit_cash(self, amount):
        print(f"Counting cash: ${amount}")
        self.ledger.append(f"Deposit: ${amount}")

    # Task 2: Loan Officer Duty
    def evaluate_mortgage(self, credit_score, income):
        print("Evaluating mortgage approval...")
        return credit_score > 700 and income > 50000

    # Task 3: Fraud Investigation
    def investigate_fraud(self, transaction_id):
        print(f"Investigating fraud for transaction {transaction_id}...")
        

def main():
    generalist = BankGeneralist()
    generalist.deposit_cash(500)
    print(generalist.evaluate_mortgage(720, 60000))
    generalist.investigate_fraud("TXN1001")


main()

File: SOLID/SRP/Violation/Bankapp.ts
class BankGeneralist {
    private ledger: string[] = [];

    // Task 1: Cash Handling
    handleCashDeposit(amount: number) {
        console.log(`Counting $${amount} in cash...`);
        this.ledger.push(`Deposit: $${amount}`);
    }

    // Task 2: Loan Evaluation
    evaluateMortgage(creditScore: number, income: number): boolean {
        console.log("Evaluating mortgage eligibility...");
        return creditScore > 700 && income > 50000;
    }

    // Task 3: Fraud Investigation
    investigateFraud(transactionId: string) {
        console.log(`Investigating fraud for transaction ${transactionId}...`);
    }
}

function main() {
    const bankEmployee = new BankGeneralist();

    bankEmployee.handleCashDeposit(500);
    console.log(bankEmployee.evaluateMortgage(750, 60000));
    bankEmployee.investigateFraud("TXN99887");
}

main();
